lib/helpers/add-one.js

```diff
@@ -0,0 +1,43 @@
+var clone = require('lodash/clone')
+var PouchDBErrors = require('pouchdb-errors')
+var Promise = require('lie')
+var uuid = require('pouchdb-utils').uuid
+
+var addTimestamps = require('../utils/add-timestamps')
+
+module.exports = function addOne (state, doc, prefix) {
+  if (typeof doc !== 'object') {
+    return Promise.reject(PouchDBErrors.NOT_AN_OBJECT)
+  }
+
+  doc = clone(doc)
+
+  if (!doc._id) {
+    doc._id = uuid()
+  }
+
+  if (prefix) {
+    doc._id = prefix + doc._id
+  }
+
+  delete doc.hoodie
+
+  return state.db.put(addTimestamps(doc))
+
+  .then(function (response) {
+    doc._id = response.id
+    doc._rev = response.rev
+    return doc
+  })
+
+  .catch(function (error) {
+    if (error.status === 409) {
+      var conflict = new Error('Object with id "' + doc._id + '" already exists')
```

Comment:
we had the idea a while ago to have a central repository where we keep all our errors, in an effort to normalise them across all repositories and also client & server: https://github.com/gr2m/milestones/issues/95

I think for now we can keep it as is, it’s just two instances of a missing and conflict error respectively